src/pipecat/services/aws_nova_sonic/aws.py

```diff
@@ -0,0 +1,1002 @@
+                            # Handle the LLM completion ending
+                            await self._handle_completion_end_event(event_json)
+
+        except Exception as e:
+            logger.error(f"{self} error processing responses: {e}")
+            if self._wants_connection:
+                await self.reset_conversation()
+
+    async def _handle_completion_start_event(self, event_json):
+        # print("[pk] completion start")
+        pass
+
+    async def _handle_content_start_event(self, event_json):
+        content_start = event_json["contentStart"]
+        type = content_start["type"]
+        role = content_start["role"]
+        generation_stage = None
+        if "additionalModelFields" in content_start:
+            additional_model_fields = json.loads(content_start["additionalModelFields"])
+            generation_stage = additional_model_fields.get("generationStage")
+
+        # Bookkeeping: track current content being received
+        content = CurrentContent(
+            type=ContentType(type),
+            role=Role(role),
+            text_stage=TextStage(generation_stage) if generation_stage else None,
+            text_content=None,
+        )
+        self._content_being_received = content
+
+        # print(f"[pk] content start: {content}")
+        # if content.role == Role.ASSISTANT:
+        #     print(f"[pk] assistant content start: {content}")
+
+        if content.role == Role.ASSISTANT:
+            if content.type == ContentType.AUDIO:
+                # Note that an assistant response can comprise of multiple audio blocks
+                if not self._assistant_is_responding:
+                    # The assistant has started responding.
+                    self._assistant_is_responding = True
+                    await self._report_assistant_response_started()
+
+    async def _handle_text_output_event(self, event_json):
+        # This should never happen
+        if not self._content_being_received:
+            return
+        content = self._content_being_received
+
+        text_content = event_json["textOutput"]["content"]
+        # print(f"[pk] text output. content: {text_content}")
+        # if content.role == Role.ASSISTANT:
+        #     print(f"[pk] assistant text output. content: {text_content}")
+
+        # Bookkeeping: augment the current content being received with text
+        # Assumption: only one text content per content block
+        content.text_content = text_content
+
+    async def _handle_audio_output_event(self, event_json):
+        # This should never happen
+        if not self._content_being_received:
+            return
+
+        # Get audio
+        audio_content = event_json["audioOutput"]["content"]
+        # print(f"[pk] audio output. content: {len(audio_content)}")
+
+        # Push audio frame
+        audio = base64.b64decode(audio_content)
+        # TODO: make sample rate + channels (used in multiple places) consts
```

Comment:
👍

Reply:
ended up adding them as params: https://github.com/pipecat-ai/pipecat/pull/1704/commits/650afeed32501908108021b577e6129b8b049dc1. wdyt?

Reply:
I think it makes sense. But what if the user configures a different sample rate? Will it be supported? If not, are we going to resample it or throw an exception?